TeamCode/src/main/java/org/firstinspires/ftc/teamcode/OctoQuadLocalizer.java

```diff
@@ -0,0 +1,163 @@
+        // TODO: make sure your config has an OctoQuad device with this name
+        //   see https://ftc-docs.firstinspires.org/en/latest/hardware_and_software_configuration/configuring/index.html
+        octoquad = hardwareMap.get(OctoQuadFWv3.class, "octoquad");
+        currentPose = initialPose;
+
+        octoquad.setSingleEncoderDirection(PARAMS.odometryPortX, PARAMS.xDirection);
+        octoquad.setSingleEncoderDirection(PARAMS.odometryPortY, PARAMS.yDirection);
+
+        octoquad.setLocalizerPortX(PARAMS.odometryPortX);
+        octoquad.setLocalizerPortY(PARAMS.odometryPortY);
+
+
+
+        double mmPerTick = 25.4 * inPerTick;
+        octoquad.setLocalizerCountsPerMM_X((float) (1 / mmPerTick));
+        octoquad.setLocalizerCountsPerMM_Y((float) (1 / mmPerTick));
+
+
+        // OctoQuad offset scheme is different than Pinpoint; these are intentionally switched
+        octoquad.setLocalizerTcpOffsetMM_X((float) -(mmPerTick * PARAMS.parYTicks));
+        octoquad.setLocalizerTcpOffsetMM_Y((float) (mmPerTick * PARAMS.perpXTicks));
+
+        octoquad.setLocalizerImuHeadingScalar((float) PARAMS.angularScalar);
+
+        octoquad.setLocalizerVelocityIntervalMS(25);
+
+
+
+        // Reset the localization, IMU, and encoders
+        octoquad.resetEverything();
+
+        ElapsedTime timeout = new ElapsedTime();
+        OctoQuadFWv3.LocalizerStatus currentStatus = octoquad.getLocalizerStatus();
+        while (currentStatus != OctoQuadFWv3.LocalizerStatus.RUNNING) {
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            if (timeout.seconds() > 3) {
+                Log.println(
+                        Log.WARN,
+                        "OctoQuadLocalizer",
+                        "init: Calibration timeout reached, OQ still not ready. OctoQuad reports " + currentStatus
+                );
+                RobotLog.addGlobalWarningMessage(
+                        "OctoQuad still not ready, timeout reached. OctoQuad reports " + currentStatus +
+                                " Continuing anyway, good luck!"
+                );
+                break;
+            }
+            currentStatus = octoquad.getLocalizerStatus();
+        }
+
+        octoquad.setLocalizerPose(
+                (int) DistanceUnit.MM.fromInches(currentPose.position.x),
+                (int) DistanceUnit.MM.fromInches(currentPose.position.y),
+                (float) currentPose.heading.toDouble()
+        );
+    }
+
+    @Override
+    public Pose2d getPose() {
+        return currentPose;
+    }
+
+    @Override
+    public void setPose(Pose2d pose) {
+        currentPose = pose;
+        octoquad.setLocalizerPose(
+                (int) DistanceUnit.MM.fromInches(currentPose.position.x),
+                (int) DistanceUnit.MM.fromInches(currentPose.position.y),
+                (float) currentPose.heading.toDouble()
+        );
```

Comment:
Interesting that it provides a `setLocalizerPose()`. I'd rather not call this and use the pinpoint scheme to avoid penalizing users that call `setPose()` often (this occasionally comes up with distance sensors and whatnot).